osmoutils/slice_helper.go

```diff
@@ -117,3 +117,38 @@ func GetRandomSubset[T any](slice []T) []T {
 	n := rand.Intn(len(slice))
 	return slice[:n]
 }
+
+// Difference returns the slice of elements that are elements of a but not elements of b.
+func Difference[T comparable](a, b []T) []T {
+	mb := make(map[T]struct{}, len(a))
+	for _, x := range b {
+		mb[x] = struct{}{}
+	}
+	diff := make([]T, 0)
+	for _, x := range a {
+		if _, found := mb[x]; !found {
+			diff = append(diff, x)
+		}
+	}
+	return diff
+}
+
+// IsDistinct checks whether provided slice contains only unique elements.
+func IsDistinct[T comparable](s []T) bool {
+	m := make(map[T]struct{}, len(s))
+	for i := range s {
+		if _, ok := m[s[i]]; ok {
+			return false
+		}
+		m[s[i]] = struct{}{}
+	}
+	return true
```

Comment:
`slices` pkg could be used here (https://cs.opensource.google/go/go/+/go1.22.1:src/slices/slices.go;l=351)

```suggestion
	return len(s) == len(slices.Compact(s))
```

Reply:
slices.Compact is quite dangerous since it modifies the underlying slice. Try out this code snippet https://go.dev/play/p/dpbxd_7OZ8z. Therefore, I wouldn't recommend using slices.Compact here.